src/List/List.js

```diff
@@ -50,14 +53,25 @@ export default class List extends Component {
      */
     component: PropTypes.oneOfType([
       PropTypes.string,
-      PropTypes.func,
+      PropTypes.func, // TODO: Why a function?
```

Comment:
We are expecting an input for the `React.createElement` function. It should be a component.
:warning: The wording I have been using is wrong. ` or a ReactElement.` we are expecting a component. That's quite some files to update #Oops.